### ISSUE TITLE
Nightly on Android does not turn off Rewards icon

### ISSUE THREAD

**Issue by pratyakshm** [brave/brave-browser#20633]
## Steps to reproduce 

   1. Go to Brave Nightly Settings > Appearance.
   2. Turn *Show Brave Rewards icon in address bar** off.
   3. It prompts to restart app for changes to take effect, hence tap on **Relaunch now**.

## Actual result 

Brave Rewards icon is not disabled from the toolbar.

## Expected result

Brave Rewards icon is disabled.

## Issue reproduces how often 

Easily reproduced

## Version/Channel Information:

- Can you reproduce this issue with the current Play Store version? No
- Can you reproduce this issue with the current Play Store Beta version? *No response*
- Can you reproduce this issue with the current Play Store Nightly version? Yes

## Device details
   - Install type (ARM, x86): ARM64
   - Device type (Phone, Tablet, Phablet): Phablet
   - Android version: 11
   
## Brave version
1.36.58

### Website problems only
- Does the issue resolve itself when disabling Brave Shields? 
- Does the issue resolve itself when disabling Brave Rewards?
- Is the issue reproducible on the latest version of Chrome?

**Comment by Miyayes** [comment]
I wonder if recent change for this setting is not propagating properly over into Android? cc: @emerick https://github.com/brave/brave-browser/issues/19695

**Comment by emerick** [comment]
Yes, I think we'll need to do a real preference migration on Android. @deeppandya I may ping you for some pointers...

**Comment by Uni-verse** [comment]
I am still able to repro this issue on `1.36.93, chromium 98.0.4758.87`  - https://github.com/brave/brave-browser/releases/tag/v1.36.96

After the browser relaunches, changes do not take affect.

**Comment by kjozwiak** [comment]
Adding `QA/Blocked` for `Android` till https://github.com/brave/brave-browser/issues/21183 is addressed/fixed. The above can also be tested alongside https://github.com/brave/brave-browser/issues/19695 & https://github.com/brave/brave-browser/issues/20646 when https://github.com/brave/brave-browser/issues/21183 is fixed.

**Comment by srirambv** [comment]
Verification passed as part of https://github.com/brave/brave-browser/issues/21183#issuecomment-1050609212. 
- Verified on both clean install and upgrade scenario rewards button is hidden or shown based on the setting
-
